Forms/SignUpForm.cs

```diff
@@ -22,53 +22,65 @@ public SignUpForm()
 
         private void SubmitButton_Click(object sender, EventArgs e)
         {
-            if (textBoxFirstName.Text == null || textBoxMiddleName.Text == null || textBoxLastName.Text == null ||
-                textBoxAddress.Text == null || textBoxEmail.Text == null || textBoxPhone.Text == null ||
-                Controls.OfType<RadioButton>().FirstOrDefault(x => x.Checked) == null)
+            try
             {
-                string message = "All fields must be fill up.";
-                string caption = "Please try again.";
-                DialogResult result = MessageBox.Show(message, caption, MessageBoxButtons.OK);
-            }
-            else
-            {
-                Customer customer = new Customer();
-                customer.Id = GenerateID.GetGeneratedID();
-                customer.FirstName = textBoxFirstName.Text;
-                customer.MiddleName = textBoxMiddleName.Text;
-                customer.LastName = textBoxLastName.Text;
-                customer.Gender = Controls.OfType<RadioButton>().FirstOrDefault(x => x.Checked).Text;
-                customer.ContactNo = textBoxPhone.Text;
-                customer.Email = textBoxEmail.Text;
-                customer.Address = textBoxAddress.Text;
-
-                if (Manager.Add(customer))
+                if (string.IsNullOrEmpty(textBoxFirstName.Text) || string.IsNullOrEmpty(textBoxMiddleName.Text) || string.IsNullOrEmpty(textBoxLastName.Text) ||
+                    string.IsNullOrEmpty(textBoxAddress.Text) || string.IsNullOrEmpty(textBoxEmail.Text) || string.IsNullOrEmpty(textBoxPhone.Text) ||
+                    Controls.OfType<RadioButton>().FirstOrDefault(x => x.Checked) == null)
                 {
-                    success = true;
-                    IForm logInForm = Application.OpenForms.OfType<LogInForm>().FirstOrDefault();
-                    ((LogInForm)logInForm).LoadData();
-
-                    IForm profileForm = new ProfileForm(customer.Id);
-                    ((ProfileForm)profileForm).MdiParent = this.ParentForm;
-                    ((ProfileForm)profileForm).Show();
-                    this.Close();
+                    string message = "All fields must be fill up.";
+                    string caption = "Please try again.";
+                    DialogResult result = MessageBox.Show(message, caption, MessageBoxButtons.OK);
                 }
                 else
                 {
-                    string message = "";
-                    string caption = "Please try again.";
-                    DialogResult result = MessageBox.Show(message, caption, MessageBoxButtons.OK);
+                    Customer customer = new Customer();
+                    customer.Id = GenerateID.GetGeneratedID();

```

Comment:
Rename GenerateID. And can you explain the reason why are you generating your ID rather than letting your database do that job?

Reply:
Since these entities will be directly use by the program, rather than creating an entity, adding it to the database and fetching it again (and this part is complicated since the program/application doesn't own the PK, it will be hard to determine the newly added record in the DB), I think this way is more efficient. Which is different for the case of Item, on which it will be just up for storage, waiting for the user for it to use.